### PR TITLE
Vend-a-tray and displaycase changes atomized

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

Ported:
https://github.com/tgstation/tgstation/pull/48994
https://github.com/tgstation/tgstation/pull/52275

## Changelog
:cl: ArcaneMusic,  ShizCalev 
add: Nanotrasen's Customer Service Division has developed a new, shielded product cage to better allow for sale of food and drink items on station, called the "Vend-A-Tray".
add: Vend-A-Trays are made with display case frames and a manipulator stock part.
fix: Fixed display cases going invisible when you put things inside of them.
fix: Fixed the pedestal for open display cases being glass colored.
/:cl:

### CODE REVIEW

**Comment by utilbird** [comment]
capitalism

File: code/game/objects/obj_defense.dm [hunk 1]
```diff
@@ -255,7 +256,7 @@ GLOBAL_DATUM_INIT(acid_overlay, /mutable_appearance, mutable_appearance('icons/e
 	qdel(src)
 
 //what happens when the obj's health is below integrity_failure level.
-/obj/proc/obj_break(damage_flag)
+/obj/proc/obj_break	(damage_flag)
```
Comment (PowerfulBacon) [inline code/game/objects/obj_defense.dm]: ?

Comment (Archanial) [reply code/game/objects/obj_defense.dm]: that's called alt tab but you missclick alt

Comment (Archanial) [reply code/game/objects/obj_defense.dm]:
```suggestion
/obj/proc/obj_break(damage_flag)
```

**Comment by BrumalLake** [comment]
Sucks to be an assistant trying to get food
People dont use this on tg, if people start using it here it'll be a dick move

File: code/modules/research/stock_parts.dm [hunk 2]
```diff
@@ -313,6 +313,12 @@ If you create T5+ please take a pass at gene_modder.dm [L40]. Max_values MUST fi
 	desc = "A large piece of equipment used to open a window into the subspace dimension."
 	materials = list(/datum/material/iron=50)
 
+/obj/item/stock_parts/card_reader
+	name = "card reader"
+	icon_state = "card_reader"
+	desc = "A small magnetic card reader, used for devices that take and transmit holocredits."
+	custom_materials = list(/datum/material/iron=50, /datum/material/glass=10)
```
Comment (francinum) [inline code/modules/research/stock_parts.dm]: Are any more uses planned for this item or is it going to go the way of the console screen, utterly useless beyond a few recipes, and not part of an upgrade tree.

Comment (Archanial) [reply code/modules/research/stock_parts.dm]: No, I didn't plan much use of it but I guess anything that takes credits can require it.

**Comment by francinum** [comment]
Get rid of the credit component.
Require an existing stock part or some non single use item

**Review by utilbird — DISMISSED** [review]

**Review by PowerfulBacon — DISMISSED** [review]

File: code/game/objects/obj_defense.dm [hunk 3]
```diff
@@ -6,20 +6,21 @@
...
+	if((resistance_flags & INDESTRUCTIBLE) || obj_integrity <= 0)
+		return
+	damage_amount = run_obj_armor(damage_amount, damage_type, damage_flag, attack_dir, armour_penetration)
+	if(damage_amount <= DAMAGE_PRECISION)
+		return
+	. = damage_amount
+	var/old_integ = obj_integrity
+	obj_integrity = max(old_integ - damage_amount, 0)
+	//BREAKING FIRST
+	if(integrity_failure && obj_integrity <= integrity_failure * max_integrity)
```
Comment (PowerfulBacon) [inline code/game/objects/obj_defense.dm]: Does this completely change how integrity failure works but doesnt update the other machines?

Comment (Archanial) [reply code/game/objects/obj_defense.dm]: If anything they will be working as originally intended

**Review by PowerfulBacon — DISMISSED** [review]

**Review by PowerfulBacon — DISMISSED** [review]

**Comment by Archanial** [comment]
?rebuild

**Review by PowerfulBacon — DISMISSED** [review]

**Comment by PowerfulBacon** [comment]
green tick for corg, I really thing the integrity failure will cause them to instantly break.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**